packages/core/http/core-http-server/src/router/utils.ts

```diff
@@ -52,3 +55,41 @@ export function getResponseValidation(
   if (typeof value === 'function') value = value();
   return isFullValidatorContainer(value) ? value.response : undefined;
 }
+
+/**
+ * Zod validation factory for Kibana route's request validation.
+ * It allows to pass a Zod schema for parameters, query and/or body validation.
+ *
+ * Example:
+ *
+ * ```ts
+ * router.versioned
+ *   .post({
+ *     access: 'public',
+ *     path: MY_URL,
+ *   })
+ *   .addVersion(
+ *     {
+ *       version: 'my-version',
+ *       validate: {
+ *         request: {
+ *           params: buildRouteValidationWithZod(MyRequestParamsZodSchema),
+ *           query: buildRouteValidationWithZod(MyRequestQueryZodSchema),
+ *           body: buildRouteValidationWithZod(MyRequestBodyZodSchema),
+ *         },
+ *       },
+ *     },
+ * ```
+ * @public
+ */
+export function buildRouteValidationWithZod<ZodSchema extends ZodType, Type = TypeOf<ZodSchema>>(
```

Comment:
> [!NOTE]
> I moved this closer to the route validation types to prevent circular references between `@kbn/zod` and `@kbn/core`

Reply:
This zod helper code should not be exported from Core. Rather redeclare `RouteValidationFunction` in `@kbn/zod` or `@kbn/zod-helpers` to avoid this circular dep.

This will leak validator implementation details to Core. Specifics about route validation are constrained to _type only_ `packages/core/http/core-http-server/src/router/route_validator.ts` with the exception of one helper: "isXSchema", e.g. `isZod`. We should keep this isolation.